controllers/workspace/devworkspace_controller.go

```diff
@@ -385,6 +388,21 @@ func getWorkspaceId(instance *devworkspace.DevWorkspace) (string, error) {
 	return "workspace" + strings.Join(strings.Split(uid.String(), "-")[0:3], ""), nil
 }
 
+// Mapping the pod to the devworkspace
+func mapPodToDW(mapObj handler.MapObject) []reconcile.Request {
+	meta := mapObj.Meta
+	labels := meta.GetLabels()
+	if _, ok := labels[constants.WorkspaceIDLabel]; !ok {
+		return nil
+	}
+	return []reconcile.Request{
+		{NamespacedName: types.NamespacedName{
+			Name:      labels[constants.WorkspaceNameLabel],
```

Comment:
We may want to verify that the label is not empty. I don't know what exactly happens when we enqueue a reconcile for an empty name.

Since the predicate should filter pod events when the pod does not have the workspaceID label, maybe we can replace the check above?

```
if _, ok := labels[constants.WorkspaceNameLabel]; !ok {
  return nil
}
```

Reply:
Yup! Makes sense! I've changed it!